### DISCUSSION TITLE
Tutorial / explanation of codebase

### DISCUSSION THREAD

**Discussion by onx2** [Sixze/ALS-Refactored#497] [Q&A]
It would be awesome to have a video series about this project, explaining why and how you chose different approaches. What you have here is amazing but pretty complex, and since I'm new to game development, it's a bit tough to grasp the entirety of it. I bet others in the community would really appreciate a walk-through too.

I'm especially interested in the network optimizations that have been done 😄 

Thanks for considering and for releasing this free!

**Comment by murataka** [comment]
you need to play with codebase yourself to understand the codebase. that is because , every one is in different level , so the documentation may not cover your cases even if it was written very detailed. that may be one of the reasons one may not spend time to make a more detailed doc , covering everything.

the network optimizations done so far mostly are not specific to this codebase , but to understand what has been done in this codebase for network optimizations, you need to understand unreal's network replication model(s) in detail first. 

as far as i see ,the discussions made here on specific titles would later become as a larger documentation library . 

still , if you want to prepare a detailed documentation , you should try to make one if you think it will be feasible. I'm sure people will help you to make corrections about the mistakes you make while preparing the documentation.
